src/main/java/Duke.java

```diff
@@ -1,10 +1,183 @@
+        ArrayList<String> inputArguments = new ArrayList<>();
+        String[] splitInput = userInput.split(" ");
+        String userCommand = splitInput[0];
+        inputArguments.add(userCommand);
+
+        // Build arguments (if any): description and command option (at/by) information (only 1 is expected)
+        String commandOption = "";
+        StringBuilder sbArgument = new StringBuilder();
+        for (int i = 1; i < splitInput.length; i++) {
+            boolean isFirstCommandOption = commandOption.isEmpty() && splitInput[i].startsWith("/");
+            if (isFirstCommandOption) {
+                String description = sbArgument.toString().stripTrailing();
+                commandOption = splitInput[i];
+                inputArguments.add(description);
+                inputArguments.add(commandOption);
+                sbArgument.setLength(0);
+                continue;
+            }
+
+            sbArgument.append(splitInput[i]).append(" ");
+        }
+
+        // Depending on user input, it could be description or task index or at/by information or empty
+        String lastArgument = sbArgument.toString().stripTrailing();
+        inputArguments.add(lastArgument);
+        return inputArguments.toArray(new String[0]);
+    }
+
+    private static void addTask(ArrayList<Task> taskList, Task newTask) {
+        pendingTaskCount++;
+        taskList.add(newTask);
+        System.out.println(" Gotcha! I've added this task: ");
+        System.out.println("\t" + newTask.toString());
+        System.out.println(" There's currently " + taskList.size() + " task(s) in the list." );
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        greetUser();
+
+        boolean shouldExit = false;
+        ArrayList<Task> taskList = new ArrayList<>();
+        Scanner scanner = new Scanner(System.in);
+        while (!shouldExit) {
+            System.out.print(System.lineSeparator() + ">> ");
+            String userInput = scanner.nextLine();
+            System.out.println(HORIZONTAL_LINE);
+
+            String[] inputArguments = parseUserInput(userInput);
+            String userCommand = inputArguments[0];
+            String description = inputArguments[1];
+            boolean isValidDescription, isValidOption, isValidOptionInformation, isValidCommand;
+            switch (userCommand) {
+            case COMMAND_BYE:
+                System.out.println(" Bye-bye. Hope to see you again soon!");
+                shouldExit = true;
+                break;
+
+            case COMMAND_LIST:
+                if (taskList.isEmpty()) {
+                    System.out.println(" Uhh.. It's empty..");
```

Comment:
I like how you handled the case where the list is empty! :)

Reply:
Thank you.